cpp/src/io/comp/gpuinflate.cu

```diff
@@ -1063,39 +1065,44 @@ __global__ void __launch_bounds__(NUMTHREADS)
     }
     __syncthreads();
     if (!state->err && (state->btype == 1 || state->btype == 2)) {
+      // Initializes lookup tables (block wide)
       init_length_lut(state, t);
       init_distance_lut(state, t);
 #if ENABLE_PREFETCH
+      // Initialize prefetcher
       init_prefetcher(state, t);
 #endif
       if (t < BATCH_COUNT) { state->x.batch_len[t] = 0; }
       __syncthreads();
       // decode data until end-of-block code
       if (t < 1 * 32) {
-        // WARP0
+        // WARP0: decode variable-length symbols
```

Comment:
Isn't this done by `thread0` only, not the whole warp? I think it would be nice to indicate which parts are done by the whole warp, vs one thread.

Reply:
The comment associated with the first branch `if (t < 1*32)` is warp selection, so the whole warp is involved. The subsequent `if (!t)` branch selects the first thread in the warp, so only thread0 ends up doing any work (will add a second comment above the `if (!t)` condition to make this clearer).

Reply:
Added more comments to debrotli: pretty much everything in debrotli is thread0-only (most of the code is decoding the metablock header, and all VLD is done by thread0 throughout, threads 1..31 are only used for LZ77, and threads 32..255 are only used for uncompressed memcpy). Building the huffman lookup tables could take advantage of multiple threads, but it would add quite a bit of complexity due to the place where the tables are built (there are huffman tables used to decode other huffman tables, unlike inflate, we can't defer the LUT creation until after header is parsed). It's also a relatively small part of the overall decompression time iirc.

Reply:
Also added a big section describing the brotli custom heap (above the function that returns the max heap size, which looks like a scratch space to the caller).